src/animals/Crocodile.ts

```diff
@@ -11,19 +12,22 @@ export default class Crocodile extends Reptile {
     }
     this.length = length;
   }
-  
-  
-  giveBirth(): [string, number, boolean,string][] {
-    const crocodilesTuples = super.giveBirth();
-    for (var crocodileTuple of crocodilesTuples) {
 
-      const babySnake = new Crocodile(crocodileTuple[0], 0, crocodileTuple[2], 10);
-      families.forEach((family) => {
-        if (family.name === this.home) {
-          family.addAnimal(babySnake, true);
-        }
+
+  public giveBirth(): void {
+    const candidateCrocks: AnimalCandidate[] = super.giveBirth() as AnimalCandidate[];
+    let length: number;
+    if (candidateCrocks) {
+      candidateCrocks.forEach(crockObjectInfo => {
+        length = Math.floor(Math.random() * 5);
```

Comment:
chenged it to length = Math.floor(Math.random() * 4) +1;  so the length cannot be more than 4, because it doesn make sence for the baby to be 9m long, for instance